frontends/web/src/routes/settings/select-explorer.tsx

```diff
@@ -0,0 +1,152 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useTranslation } from 'react-i18next';
+import { i18n } from '../../i18n/i18n';
+import { Guide } from '../../components/guide/guide';
+import { Entry } from '../../components/guide/entry';
+import { Button, ButtonLink } from '../../components/forms';
+import { Header } from '../../components/layout';
+import { BlockExplorers } from './block-explorers';
+import * as backendAPI from '../../api/backend';
+import { apiGet } from '../../utils/request';
+import { getConfig, setConfig } from '../../utils/config';
+import { CoinCode } from '../../api/account';
+import { MobileHeader } from './components/mobile-header';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+
+type TBlockExplorer = { name: string; url: string };
+type TSelectionData = Record<CoinCode, TBlockExplorer[]>;
+
+export const SelectExplorerSettings = () => {
+  const { t } = useTranslation();
+
+  const initialConfig = useRef<any>();
+  const [config, setConfigState] = useState<any>();
+
+  const [supportedCoins, setSupportedCoins] = useState<backendAPI.ICoin[]>([]);
+  const [allSelections, setAllSelections] = useState<TSelectionData>();
+
+  const [saveDisabled, setSaveDisabled] = useState(true);
+
+  const loadConfig = () => {
+    getConfig().then(setConfigState);
+  };
+
+
+  const updateConfigState = useCallback((newConfig: any) => {
+    if (JSON.stringify(initialConfig.current) !== JSON.stringify(newConfig)) {
+      setConfigState(newConfig);
+      setSaveDisabled(false);
+    } else {
+      setSaveDisabled(true);
+    }
+  }, []);
+
+  const handleChange = useCallback((selectedTxPrefix: string, coin: CoinCode) => {
+    if (config.backend[coin]) {
+      // if you select another explorer and then change it back to the previous state,
+      // save will be enabled...it would be nicer if we know the initial state and enable
+      // might refactor later but this is ok imo.
+      if (config.backend[coin].blockExplorerTxPrefix !== selectedTxPrefix) {
+        config.backend[coin].blockExplorerTxPrefix = selectedTxPrefix;
+        // setSaveDisabled(false);
+        // setConfigState(config);
+        updateConfigState(config);
+      }
+    }
+  }, [config, updateConfigState]);
+
+  const save = async () => {
+    setSaveDisabled(true);
+    await setConfig(config);
+    initialConfig.current = await getConfig();
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const coins = await backendAPI.getSupportedCoins();
+      const allExplorerSelection = await apiGet('available-explorers');
```

Comment:
make an API call wrapper in api/backend.ts, and a definition of the return type.